src/routes/mod.rs

```diff
@@ -1,67 +1,30 @@
 use atom_syndication::{ContentBuilder, Entry, EntryBuilder, LinkBuilder, Person, PersonBuilder};
 use rocket::{
-    http::uri::{FromUriParam, UriDisplay},
-    response::NamedFile
-};
-use std::{
-    fmt,
-    path::{Path, PathBuf}
+    http::RawStr,
+    request::FromFormValue,
+    response::{Content, NamedFile},
 };
+use std::path::{Path, PathBuf};
 
 use plume_models::{Connection, posts::Post};
 
-macro_rules! may_fail {
-    ($account:expr, $expr:expr, $template:expr, $msg:expr, | $res:ident | $block:block) => {
-        {
-            let res = $expr;
-            if res.is_some() {
-                let $res = res.unwrap();
-                $block
-            } else {
-                Template::render(concat!("errors/", $template), json!({
-                    "error_message": $msg,
-                    "account": $account
-                }))
-            }
-        }
-    };
-    ($account:expr, $expr:expr, $msg:expr, | $res:ident | $block:block) => {
-        may_fail!($account, $expr, "404", $msg, |$res| {
-            $block
-        })
-    };
-    ($account:expr, $expr:expr, | $res:ident | $block:block) => {
-        may_fail!($account, $expr, "", |$res| {
-            $block
-        })
-    };
-}
-
 const ITEMS_PER_PAGE: i32 = 12;
 
-#[derive(FromForm)]
-pub struct Page {
-    page: i32
-}
-
-impl UriDisplay for Page {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "page={}", &self.page as &UriDisplay)
-    }
-}
+pub struct Page(i32);
 
-impl FromUriParam<i32> for Page {
-    type Target = Page;
-    fn from_uri_param(num: i32) -> Page {
-        Page { page: num }
+impl<'v> FromFormValue<'v> for Page {
```

Comment:
ugh I did the same change on `search`, we will have some merge issues for literally nothing :cry: (and also in a lot of other places, but for good reasons)